### PR TITLE
aspects: optimise aspect transaction Get()

### PR DESCRIPTION
**Pull request by MiguelPires**
Apply this suggestion https://github.com/snapcore/snapd/pull/12892#discussion_r1233953868 for an optimisation to the aspect transaction's Get operation. It caches a copy of the transaction's working databag so that a new one doesn't have to be created and all the deltas re-applied to answer new Get operations. It also keeps track of which deltas were applied to the bag, so that only new changes are applied in subsequent Gets.

### CODE REVIEW

**Review by ernestl — APPROVED** [review]
LGTM

File: aspects/transaction.go [hunk 1]
```diff
@@ -65,17 +68,26 @@ func (t *Transaction) Get(path string, value interface{}) error {
...
+		t.appliedDeltas = 0
+	}
+
+	// apply new changes since the last get
+	if err := applyDeltas(t.modified, t.deltas[t.appliedDeltas:]); err != nil {
+		t.modified = nil
+		t.appliedDeltas = 0
+		return err
 	}
+	t.appliedDeltas = len(t.deltas)
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply aspects/transaction.go:88]: It's so we only apply the deltas that come after it (line 83)

[the remark this answers is not in the document]
Comment (mvo5) [reply aspects/transaction.go:88]: Aha, nice - silly me

**Review by mvo5 — APPROVED** [review]
Thank you!

**Pull request merged**